plugins/catalog/src/components/EntityPeekCard/EntityPeekCard.tsx

```diff
@@ -0,0 +1,162 @@
+import useAsync from 'react-use/lib/useAsync';
+import { usePopupState } from 'material-ui-popup-state/hooks';
+import { useNavigate } from 'react-router';
+import { Progress } from '@backstage/core-components';
+
+export type EntityPeekCardProps = {
+  entityRefs: CompoundEntityRef[];
+};
+
+const useStyles = makeStyles(() => ({
+  paper: { width: 320, maxWidth: '100%' },
+}));
+
+const UserGroupMenuItems = ({
+  entity,
+  popupState,
+}: {
+  entity: UserEntity | GroupEntity;
+  popupState: InjectedProps;
+}) => {
+  return (
+    <>
+      {entity.spec.profile?.email && (
+        <MenuItem onClick={popupState.close}>
+          <Link href={`mailto: ${entity.spec.profile.email}`}>
+            <ListItemIcon>
+              <EmailIcon fontSize="small" />
+            </ListItemIcon>
+            <ListItemText>
+              Send Email to {entity.spec.profile?.displayName}
+            </ListItemText>
+          </Link>
+        </MenuItem>
+      )}
+    </>
+  );
+};
+
+const EntityMenu = ({
+  popupState,
+  entityRefs,
+}: {
+  popupState: InjectedProps;
+  entityRefs: CompoundEntityRef[];
+}) => {
+  const catalogApi = useApi(catalogApiRef);
+  const navigate = useNavigate();
+
+  const { value: entities, loading } = useAsync(async (): Promise<
+    (Entity | undefined)[] | undefined
+  > => {
+    if (popupState.isOpen) {
+      return Promise.all(
+        entityRefs.map(async entityRef => {
+          return await catalogApi.getEntityByRef(entityRef);
+        }),
+      );
+    }
+    return undefined;
+  }, [popupState.isOpen]);
+
+  const onDetailsClick = useCallback(
+    (entity: Entity) => {
+      popupState.close();
+      navigate(
+        `/catalog/${entity.metadata.namespace}/${entity.kind.toLocaleLowerCase(
+          'en-US',
+        )}/${entity.metadata.name}`,
+      );
+    },
+    [navigate, popupState],
+  );
+
+  if (loading) {
+    return <Progress />;
+  }
+
+  return (
+    <Menu {...bindMenu(popupState)}>
+      {entities?.map(entity => {
+        if (entity && (isUserEntity(entity) || isGroupEntity(entity))) {
+          return <UserGroupMenuItems entity={entity} popupState={popupState} />;
+        }
+        return <></>;
+      })}
+      <Divider />
+      {entities?.map(entity => {
+        if (entity) {
+          return (
+            <MenuItem onClick={() => onDetailsClick(entity)}>
+              <ListItemIcon>
+                <KeyboardArrowDownIcon fontSize="small" />
+              </ListItemIcon>
+              <ListItemText>Details</ListItemText>
+            </MenuItem>
+          );
+        }
+        return <></>;
+      })}
+    </Menu>
+  );
+};
+
+export const EntityPeekCard = ({ entityRefs }: EntityPeekCardProps) => {
+  const classes = useStyles();
+  const popupState = usePopupState({
+    variant: 'popper',
+    popupId: 'demoMenu',
+    disableAutoFocus: true,
+  });
+
+  return (
+    <div>
+      <Button {...bindTrigger(popupState)} endIcon={<KeyboardArrowDownIcon />}>
```

Comment:
Im not terribly happy with the look of this button.

Reply:
![image](https://user-images.githubusercontent.com/553697/205346638-6c774027-6a52-4654-8891-4cececbb1dfd.png)